pkg/di/ir/program.go

```diff
@@ -0,0 +1,162 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package ir
+
+import "github.com/DataDog/datadog-agent/pkg/network/go/bininspect"
+
+// An ID corresponding to an instance of a program.
+type ProgramID uint32
+
+// An ID corresponding to a type in a program.
+type TypeID uint32
+
+// An ID corresponding to an event output by the program.
+// It is used to identify events as they are communicated over the ring buffer.
+type EventID uint32
+
+// A program defined the information needed to generate an eBPF program
+// for a set of probes as they apply to a specific binary. This structure
+// is used to drive both eBPF program generation and the interpretation of
+// data that comes out of the generated program.
+type Program struct {
+
+	// ID is the ID of the program. This is used to identify messages
+	// from this program as they are communicated over the ring buffer.
+	ID ProgramID
+
+	// Probes are the set of probes that comprise the program.
+	Probes []*Probe
+
+	// Subprograms are a list of subprograms that will be probed.
+	Subprograms []*Subprogram
```

Comment:
Do Probes always map 1:1 to subprograms? I see you have a reference to Probe in Subprogram but not the other way around.

Reply:
There can be many probes configured that target the same Subprogram (read: function).  A `Probe` corresponds to exactly one `Subprogram`, unless I misunderstand how `Where` works in the API. A `Probe` here corresponds to the `Probe` in the API like [here](https://github.com/DataDog/dd-go/blob/9bbe4002a415fd613c13ac8c986dc04c86211b22/remote-config/pkg/products/livedebugging/domain.go#L73-L88). There's a reference `Probe` to `Subprogram` so it is exactly "the other way around".